features/all-positions/PositionActionsDialog.tsx

```diff
@@ -0,0 +1,583 @@
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+  const { latestPrice } = PriceFeed.useContainer();
+  const {
+    symbol: tokenSymbol,
+    balance: tokenBalance,
+    allowance: tokenAllowance,
+    setMaxAllowance: setMaxTokenAllowance,
+  } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { activeSponsors } = EmpSponsors.useContainer();
+  const [tabIndex, setTabIndex] = useState<string | null>("deposit");
+  const [collateralToDeposit, setCollateralToDeposit] = useState<string>("");
+  const [minCollPerToken, setMinCollPerToken] = useState<string>("");
+  const [maxCollPerToken, setMaxCollPerToken] = useState<string>("");
+  const [maxTokensToLiquidate, setMaxTokensToLiquidate] = useState<string>("");
+  const [deadline, setDeadline] = useState<string>("");
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    priceIdentifier: priceId,
+    collateralRequirement: collReq,
+    currentTime,
+  } = empState;
+
+  const setDialogTab = (
+    event: MouseEvent<HTMLElement>,
+    newAlignment: string | null
+  ) => {
+    setTabIndex(newAlignment);
+  };
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String) => {
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+  if (
+    props.selectedSponsor &&
+    activeSponsors[props.selectedSponsor] &&
+    latestPrice !== null &&
+    collReq
+  ) {
+    const collBalanceNum = collBalance || 0;
+    const tokenBalanceNum = tokenBalance || 0;
+    const finalFeeNum = finalFee || 0;
+    const collateralToDepositNum = Number(collateralToDeposit) || 0;
+    const maxTokensToLiquidateNum = Number(maxTokensToLiquidate) || 0;
+
+    const sponsorPosition = activeSponsors[props.selectedSponsor];
+
+    const pendingWithdrawTimeRemaining =
+      Number(sponsorPosition.withdrawalTimestamp) - Number(currentTime);
+
+    const pendingWithdrawTimeString =
+      pendingWithdrawTimeRemaining && pendingWithdrawTimeRemaining > 0
+        ? Math.max(0, Math.floor(pendingWithdrawTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((pendingWithdrawTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (pendingWithdrawTimeRemaining % 3600) % 60)
+        : "None";
+
+    const pendingTransferTimeRemaining =
+      Number(sponsorPosition.TransferTimestamp) - Number(currentTime);
+
+    const pendingTransferTimeString =
```

Comment:
```suggestion
    const prettyTimeRemaining = (timeRemaining: number) => {
         return timeRemaining > 0
              ? Math.max(0, Math.floor(timeRemaining / 3600)) +
                 ":" +
                 Math.max(0, Math.floor((timeRemaining % 3600) / 60)) +
                 ":" +
                 Math.max(0, (timeRemaining % 3600) % 60)
              : "None";
    }
    const pendingTransferTimeString = prettyTimeRemaining(pendingTransferTimeRemaining)
    const pendingWithdrawTimeString = prettyTimeRemaining(pendingWithdrawTimeRemaining)
```

Might as well not reuse this code block. We use in other modules as well and we can eventually break into its own module